core/iwasm/libraries/lib-wasi-threads/tid_allocator.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef _TID_ALLOCATOR_H
+#define _TID_ALLOCATOR_H
+
+#include "platform_common.h"
+
+#define TID_ALLOCATOR_INIT_SIZE CLUSTER_MAX_THREAD_NUM
+enum {
+    TID_MIN = 1,
+    TID_MAX = 0x1FFFFFFF
+}; // Reserved TIDs (WASI specification)
+
+/* Stack data structure to track available thread identifiers */
+typedef struct tidAllocator TidAllocator;
+
+TidAllocator *
+tid_allocator_init();
```

Comment:
I think slightly better model in this particular case would be to leave allocation to the caller; i.e. the API would be something like:
```
bool tid_allocator_init(TidAllocator *);
void tid_allocator_deinit(TidAllocator *);
```
so then the caller can decide whether to allocate `tid_allocator` on the stack/in the data segment or on the heap (and in our case here, most likely we'd like to avoid dynamic allocations.

Reply:
good point, I'll have to expose the `TidAllocator` struct in the header, but it's better in our case so we avoid the dynamic allocation